### PR TITLE
Extends tests cases to support tenants

### PR DESCRIPTION
**Pull request by nalakas**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/NewCopyWithDefaultVersion.java [hunk 1]
```diff
@@ -72,8 +72,11 @@ public NewCopyWithDefaultVersion(TestUserMode userMode) {
 
     @BeforeClass(alwaysRun = true)
     public void setEnvironment() throws Exception {
-        super.init();
-        restAPIPublisher = new RestAPIPublisherImpl();
+        super.init(userMode);
+        String userName = gatewayContextWrk.getContextTenant().getTenantAdmin().getUserNameWithoutDomain();
+        String password = gatewayContextWrk.getContextTenant().getTenantAdmin().getPassword();
+        String tenantDomain = gatewayContextWrk.getContextTenant().getDomain();
+        restAPIPublisher = new RestAPIPublisherImpl(userName, password, tenantDomain);
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/NewCopyWithDefaultVersion.java]: Do we need this? This is already done in the Super class.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/SameVersionAPITestCase.java [hunk 2]
```diff
@@ -69,8 +69,11 @@ public SameVersionAPITestCase(TestUserMode userMode) {
 
     @BeforeClass(alwaysRun = true)
     public void setEnvironment() throws Exception {
-        super.init();
-        restAPIPublisher = new RestAPIPublisherImpl();
+        super.init(userMode);
+        String userName = gatewayContextWrk.getContextTenant().getTenantAdmin().getUserNameWithoutDomain();
+        String password = gatewayContextWrk.getContextTenant().getTenantAdmin().getPassword();
+        String tenantDomain = gatewayContextWrk.getContextTenant().getDomain();
+        restAPIPublisher = new RestAPIPublisherImpl(userName, password, tenantDomain);
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/SameVersionAPITestCase.java]: Same here. We will not need this. We can use the variables defined in the super class.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/NewCopyWithDefaultVersion.java (hunk 1 above)
Comment (nalakas) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/NewCopyWithDefaultVersion.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/SameVersionAPITestCase.java (hunk 2 above)
Comment (nalakas) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/SameVersionAPITestCase.java]: fixed

**Pull request merged**